it/controllers/benefits/reimbursed/VouchersBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,503 @@
+    val maxAmountErrorText: String = "The amount for vouchers or credit cards must be less than £100,000,000,000"
+  }
+
+  object ExpectedIndividualEN extends SpecificExpectedResults {
+    val youCanText: String = "You can find this information on your P11D form in section C, box 12."
+  }
+
+  object ExpectedIndividualCY extends SpecificExpectedResults {
+    val youCanText: String = "You can find this information on your P11D form in section C, box 12."
+  }
+
+  object ExpectedAgentEN extends SpecificExpectedResults {
+    val youCanText: String = "You can find this information on your client’s P11D form in section C, box 12."
+  }
+
+  object ExpectedAgentCY extends SpecificExpectedResults {
+    val youCanText: String = "You can find this information on your client’s P11D form in section C, box 12."
+  }
+
+  private def benefits(reimbursedCostsVouchersAndNonCashModel: ReimbursedCostsVouchersAndNonCashModel): BenefitsViewModel =
+    BenefitsViewModel(carVanFuelModel = Some(fullCarVanFuelModel), accommodationRelocationModel = Some(fullAccommodationRelocationModel),
+      travelEntertainmentModel = Some(fullTravelOrEntertainmentModel), utilitiesAndServicesModel = Some(fullUtilitiesAndServicesModel),
+      medicalChildcareEducationModel = Some(fullMedicalChildcareEducationModel), incomeTaxAndCostsModel = Some(fullIncomeTaxAndCostsModel),
+      isUsingCustomerData = true, isBenefitsReceived = true, reimbursedCostsVouchersAndNonCashModel = Some(reimbursedCostsVouchersAndNonCashModel))
+
+  private val benefitsWithNoBenefitsReceived: Option[BenefitsViewModel] = Some(BenefitsViewModel(isUsingCustomerData = true))
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, SpecificExpectedResults]] = Seq(
+    UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+    UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+    UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+    UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY))
+  )
+
+  ".show" should {
+    userScenarios.foreach { user =>
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+        import Selectors._
+        import user.commonExpectedResults._
+        "render the vouchers benefits amount page without pre-filled form" which {
+          lazy val result: WSResponse = {
+            dropEmploymentDB()
+            authoriseAgentOrIndividual(user.isAgent)
+            val model = fullReimbursedCostsVouchersAndNonCashModel.copy(vouchersAndCreditCards = None)
+            insertCyaData(employmentUserData(isPrior = true, cyaModel(Some(benefits(model)))), userRequest)
+            userDataStub(userData(fullEmploymentsModel(hmrcEmployment = Seq(employmentDetailsAndBenefits(fullBenefits)))), nino, taxYearEOY)
```

Comment:
would vouchersAndCreditCards need to be empty in the model?

Reply:
it is empty in the model? it passes in the model on like 143 where its set to none

Reply:
you're right. I missed that part